### DISCUSSION TITLE
Pact testing issue in Windows machine

### DISCUSSION THREAD

**Discussion by premNtt** [pact-foundation/pact-js#1178] [Q&A]
While running the pact testing in my Nextjs spa app, am facing the pact testing issue in windows machine error : pact-node@10.17.2: Pact Binary Error: ..\node_modules@pact-foundation\pact\node_modules@pact-foundation\pact-node\standalone\win32-1.88.83\pact\lib\ruby\bin.real\ruby.exe is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher.

am using https://github.com/pact-foundation versions are
"@pact-foundation/pact": "9.17.2",
"@pact-foundation/pact-node": "10.17.1",

and also am using Windows 10 Enterprise, 64-bit operating system, x64-based processor
Please help on this issue

**Comment by portyu9** [comment]
Hi @premNtt 

The important part of the error is this:

```text
...\pact-node\standalone\win32-1.88.83\pact\lib\ruby\bin.real\ruby.exe
is not compatible with the version of Windows you're running
```

That comes from the **old Pact JS v9 architecture**, which bundled the Ruby Pact standalone binaries.

Your versions:

```json
"@pact-foundation/pact": "9.17.2",
"@pact-foundation/pact-node": "10.17.1"
```

are now several major generations behind the current implementation.

The preferred fix today is **not to troubleshoot that bundled `ruby.exe`**. Upgrade Pact JS.

Pact JS v10 replaced the Ruby shared core with the Rust shared core, and v12+ provides prebuilt native dependencies.

Also remove the direct `pact-node` dependency unless you have some separate legacy code that specifically requires it:

```bash
npm uninstall @pact-foundation/pact-node
```

Then upgrade Pact JS to a release supported by your Node version.

For a current Node environment:

```bash
npm install --save-dev @pact-foundation/pact@latest
```

Current Pact JS uses:

```text
@pact-foundation/pact
        ↓
@pact-foundation/pact-core
        ↓
Rust Pact core
```

rather than launching the old bundled Ruby executable shown in your error.

If you're upgrading directly from v9, check the Pact JS migration guides because there were API/lifecycle changes. In particular, newer consumer APIs generally use `executeTest()` rather than manually managing the old long-running mock-server lifecycle.

Also note that very recent Pact JS releases require a modern Node version, so upgrade Node as part of the migration if necessary rather than forcing the newest Pact version into an old Next.js/Node environment.

For the specific Windows error in this thread, I would therefore **upgrade away from Pact JS 9 rather than trying to repair or replace its bundled Ruby executable**.

Please mark this answer as accepted if it helped, thank you.
